services/datamanager/builtin/builtin.go

```diff
@@ -433,6 +437,28 @@ func (svc *builtIn) Reconfigure(
 				// do not have the resource right now
 				continue
 			}
+
+			captureFrequencyKey := fmt.Sprintf("%s-capture-frequency", resConf.Name.String())
```

Comment:
Since the name can actually be shared across different components and only IDs are unique, we created a unique identifier for what can constitute a "collector" or unique capture: the [resourceMethodMetadata](https://github.com/viamrobotics/rdk/blob/8d8bc93b5f9c1c4616945d5f7c6aa3c646755130/services/datamanager/builtin/builtin.go#L443-L447).

This resourceMethodMetadata is composed of the component name (e.g. 'my-arm'), API (e.g. the full 'arm' [type](https://github.com/viamrobotics/rdk/blob/main/resource/api.go#L27-L32)), method (e.g. 'GetPosition'), and parameters (even if the name/method is the same, something like `board` can have different parameters on the same method which get different results that we want to capture separately). We use this unique identifier as a key in [newCollectors](https://github.com/viamrobotics/rdk/blob/8d8bc93b5f9c1c4616945d5f7c6aa3c646755130/services/datamanager/builtin/builtin.go#L429), so would want to reuse the same thing here — can move up the instantiation of that variable to here and reuse it below in the if-block.